### DISCUSSION TITLE
[V3 Beta]

### DISCUSSION THREAD

**Discussion by Larry-Home** [awcodes/filament-curator#286] [Q&A] [answered]
I recently decided to try the upgrade to Livewire/Filamant 3 and for the most part has been good.  Having an issue here though with a form using a field which has not changed since v2 like:
```
CuratorPicker::make('media_id')
                    ->required()
                    ->disk('public')
                    ->directory('images/events')
                    ->limitToDirectory()
                    ->preserveFilenames()
                    ->imageResizeTargetHeight('288'),
```

Produces the error:

Route [filament.admin.resources.media.edit] not defined.

If I comment out the field above the form loads / functions so it definitely is this field being involved.  

What might be 100% related is if I try and run php artisan curator:upgrade it generates the following error:

```
Backing up table to `media_tmp`...

   BadMethodCallException 

  Method Illuminate\Foundation\Application::getTable does not exist.

  at vendor/laravel/framework/src/Illuminate/Macroable/Traits/Macroable.php:112
    108▕      */
    109▕     public function __call($method, $parameters)
    110▕     {
    111▕         if (! static::hasMacro($method)) {
  ➜ 112▕             throw new BadMethodCallException(sprintf(
    113▕                 'Method %s::%s does not exist.', static::class, $method
    114▕             ));
    115▕         }
    116▕ 

  i   Bad Method Call: Did you mean Illuminate\Foundation\Application::getLocale() ? 

      +13 vendor frames 

  14  artisan:35
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
```

Going to keep working on the resolution here, but thought maybe others had a path to get this resolved quickly.  Thanks!

**Comment by awcodes** [comment, marked as answer]
I am aware of the bug in the upgrade script migration. Hopefully, will get that addressed soon.

Not sure why you wouldn't be getting a route though, did you add the plugin to your panel, that should auto detect the resources and the routes.

**Reply by Larry-Home** [reply]
As a hint for anybody reaching this while 3.0 is still beta.  Look at the included README.md with the package.  I was indeed missing the needed plugin in my panel.  As always thanks for your assistance!

**Reply by awcodes** [reply]
Pushed a release today too that should fix the upgrade script. All it's really doing though is adding a visibility column to the media table. Cheers.

**Reply by Larry-Home** [reply]
Can confirm the update makes the script function properly!

**Reply by awcodes** [reply]
Awesome. Thank you.

**Comment by GalinaBublik** [comment]
Got an error. Please help
![image](https://github.com/awcodes/filament-curator/assets/10061993/c9dcbaf6-f59c-4d1b-9eea-709c4fa771b2)

composer.json
"require": {
        "php": "^8.1",
        "alperenersoy/filament-export": "^0.3.7",
        "awcodes/filament-curator": "^2.7",
        "bezhansalleh/filament-shield": "^2.4",
        "cmdinglasan/filament-boring-avatars": "^1.1",
        "cms-multi/filament-clear-cache": "^1.0",
        "darkaonline/l5-swagger": "8.1",
        "filament/filament": "^2.17",
        "filament/forms": "^2.17",
        "filament/spatie-laravel-settings-plugin": "^2.17",
        "filament/tables": "^2.17",
        "fruitcake/laravel-cors": "^2.0.5",
        "guzzlehttp/guzzle": "^7.2",
        "jeffgreco13/filament-breezy": "^1.5",
        "laravel/framework": "^9.0",
        "laravel/passport": "11.8",
        "laravel/sanctum": "^2.14",
        "laravel/tinker": "^2.7",
        "leandrocfe/filament-apex-charts": "^1.0",
        "mohamedsabil83/filament-forms-tinyeditor": "^1.7",
        "oneplusone/cmwquery": "^1.0",
        "predis/predis": "^2.1",
        "ryangjchandler/filament-navigation": "^0.5.0",
        "stevebauman/purify": "^6.0",
        "z3d0x/filament-logger": "^0.5.6"
    },

Removed, tried re-install 
![image](https://github.com/awcodes/filament-curator/assets/10061993/ab9aef15-8c5f-4b82-a7d4-8b6c43459ae7)

**Comment by awcodes** [comment]
v3 only works with Filament v3
